R/epi_calendar.R

```diff
@@ -23,34 +23,32 @@
 #'
 #' @examples
 #' # Check if a date is within the epidemiological season
-#' epi_calendar(as.Date("2023-09-15"), start = 40, end = 20)
-#' # Expected output: "2022/2023"
+#' epi_calendar(as.Date("2023-09-15"), start = 21, end = 20)
+#' # Expected output: "2023/2024"
 #'
-#' epi_calendar(as.Date("2023-05-01"), start = 40, end = 20)
+#' epi_calendar(as.Date("2023-05-30"), start = 40, end = 20)
 #' # Expected output: "out_of_season"
 #'
-#' epi_calendar(as.Date("2023-01-15"), start = 40, end = 20)
+#' epi_calendar(as.Date("2023-01-15"), start = 21, end = 20)
 #' # Expected output: "2022/2023"
 #'
-#' epi_calendar(as.Date("2023-12-01"), start = 40, end = 20)
+#' epi_calendar(as.Date("2023-12-01"), start = 21, end = 20)
 #' # Expected output: "2023/2024"
-epi_calendar <- Vectorize(function(date, start = 40, end = 20) {
+epi_calendar <- Vectorize(function(date, start = 21, end = 20) {
   # Compute the current week
-  current_week <- as.integer(format(x = date, "%U"))
+  current_week <- as.integer(format(x = date, "%V"))
 
-  if (current_week <= start & end <= current_week) {
+  if (!(current_week >= start | current_week <= end)) {
     return("out_of_season")
   }
 
   # Compute the current year
-  current_year <- format(date, "%Y")
-  # ... and turn into integer
-  current_year_integer <- as.integer(current_year)
+  current_year <- as.integer(strftime(date, format = "%G"))
 
   if (current_week <= end) {
-    ans <- paste0(current_year_integer - 1, "/", current_year_integer)
+    ans <- paste0(current_year - 1, "/", current_year)
   } else {
-    ans <- paste0(current_year_integer, "/", current_year_integer + 1)
+    ans <- paste0(current_year, "/", current_year + 1)
   }
```

Comment:
Again, this would not work for within-year seasons. But maybe you dont want to support that. In that case, I think you should add a checkmate check that requires end < start